### DISCUSSION TITLE
Layering Images in bevy_ui

### DISCUSSION THREAD

**Discussion by leo6dev** [bevyengine/bevy#4694] [Q&A] [answered]
Hi sorry if this is the incorrect place for this but I am struggling with layering  images but it appears that when overlaying images with code like this
```rust
  use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn_bundle(UiCameraBundle::default());

    commands.spawn()
        .insert_bundle(ImageBundle {
            color: Color::WHITE.into(),
            ..default()
        })
        .insert_bundle(ImageBundle {
            color: Color::NONE.into(),
            ..default()
        });
}
```
The ```Color::WHITE``` image is not displayed even tho the other image has an alfa of 0 and should be total transperant. by removing the second image (NONE) the first (WHITE) is correctly displayed. A real world example of this is when using pngs with alfa, this was my orignal use case witch involved trying to draw 2 pngs on top of one and other.

Issue occurs on 0.7 and git main

**Comment by tim-blackbird** [comment, marked as answer]
Your second .insert_bundle() is overwriting the first call. You'll need to call commands.spawn() before each bundle to spawn 2 separate entities .

**Reply by leo6dev** [reply]
Hi is there a way to have the images overlap?
I had already tried that with this result 
![image](https://user-images.githubusercontent.com/76897751/167288636-ee9150c7-e9e1-4d64-a928-ba93241bf0f5.png)
I was hoping to have the second image overlap the first

Sorry for the fact that its not from the same example code.

Thanks

**Reply by tim-blackbird** [reply]
No worries.
Haven't played around much with the ui myself, but I think you need to make the axe a child of the frame.
I recommend taking a look at the [ui example](https://github.com/bevyengine/bevy/blob/latest/examples/ui/ui.rs) to see how you can do that.

**Reply by Geams** [reply]
You have to add the top image as a child to the bottom image, similar to this [example](https://github.com/bevyengine/bevy/blob/v0.7.0/examples/ui/ui.rs).
```

let bottom = ImageBundle{
        style: Style{
            size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
            ..default()
        },
        color: Color::WHITE.into(),
        ..default()
    };
    let top = ImageBundle{
        style: Style{
            size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
            ..default()
        },
        color: Color::rgba_u8(255,0,0,100).into(),
        ..default()
    };
 commands
    .spawn_bundle(bottom)
        .with_children(|parent|{
            parent.spawn_bundle(top);
        });
```

**Reply by leo6dev** [reply]
Thanks
